web/src/components/project/validation/validation-result.tsx

```diff
@@ -2,23 +2,25 @@ import { useState } from 'react';
 import { OverlayTrigger, Tooltip } from 'react-bootstrap';
 
 import { useValidation } from '../../../hooks/queries/useValidation';
-import { ValidationResultModal } from '../../modals/validation-result';
 import { StatusIcon } from '../../badges/status-icons';
-
+import { ValidationResultModal } from '../../modals/validation-result';
 
 type Props = {
   schemaRegistry: string | undefined;
   isValidating: boolean;
   validationResult: ReturnType<typeof useValidation>['data'];
+  shouldValidate: boolean;
 };
 
 export const ValidationResult = (props: Props) => {
-  const { validationResult, isValidating, schemaRegistry } = props;
+  const { validationResult, isValidating, schemaRegistry, shouldValidate } = props;
 
   const [validationModalIsOpen, setValidationModalIsOpen] = useState(false);
 
   let wrapperClassName = 'py-1 px-2 rounded-1 bg-opacity-10 validation-button';
-  if (isValidating) {
+  if (!shouldValidate) {
+    wrapperClassName += ' border border-warning text-warning bg-warning';
+  } else if (isValidating) {
```

Comment:
shouldValidate is returning false only so the validation button never changes color from yellow.. Color change works if you set the if statement to check for shouldValidate and not !shouldValidate..